server/channels/store/sqlstore/post_store.go

```diff
@@ -1110,17 +1097,34 @@ func (s *SqlPostStore) permanentDeleteAllCommentByUser(userId string) (err error
 // Permanently deletes all comments by user,
 // cleans up threads (removes said user from participants and decreases reply count),
 // permanent delete all root posts by user,
+// permanent delete all reactions by user,
 // and delete threads and thread memberships for those root posts
+// Note that this method can potentially take a very long time, and is not meant
+// to be called synchronously, but only via jobs.
 func (s *SqlPostStore) PermanentDeleteByUser(rctx request.CTX, userId string) error {
 	// First attempt to delete all the comments for a user
-	if err := s.permanentDeleteAllCommentByUser(userId); err != nil {
-		return err
+	for {
+		results := []postIds{}
+		err := s.GetReplica().Select(&results, "Select Id, RootId FROM Posts WHERE UserId = ? AND RootId != '' LIMIT 10", userId)
+		if err != nil {
+			return errors.Wrapf(err, "failed to fetch comment Posts with userId=%s", userId)
+		}
+
+		if len(results) == 0 {
+			break
+		}
+
+		// This creates a long transaction for each element in the results slice.
+		// That's why we limit the number of rows to 10.
+		if err := s.permanentDeleteAllCommentByUser(userId, results); err != nil {
+			return err
+		}
+
+		time.Sleep(time.Second)
```

Comment:
I am slightly confused here: we limit replies to 10 above, but then `permanentDeleteAllCommentByUser` runs `DELETE FROM Posts WHERE UserId = ? AND RootId != ''`, effectively deleting all replies. Is that what we want?

Reply:
Gah sorry. Refactoring bug. I'll adjust.